File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/Lasagna.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism;

public class Lasagna
{
    public int ExpectedMinutesInOven() => 40;
    
    public int RemainingMinutesInOven(int minutes) => ExpectedMinutesInOven() - minutes;
    
    public int PreparationTimeInMinutes(int layers) => layers * 2;
    
    public int ElapsedTimeInMinutes(int layers, int time) => layers * 2 + time;
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven(){
        return 40;
    }

    public int remainingMinutesInOven(int time){
        return expectedMinutesInOven() - time;
    }

    public int preparationTimeInMinutes(int layers){
        return layers * 2;
    }

    public int totalTimeInMinutes(int layers, int time){
        return time + preparationTimeInMinutes(layers);
    }
}